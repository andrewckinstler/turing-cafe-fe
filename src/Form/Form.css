.form-box {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
}

input {
  padding: 0;
  width: 150px;
  height: 30px;
  font-size: 1em;
  margin: 0 30px;
}

.create {
  background-color: teal;
  color: #FFF;
  width: 150px;
  height: 34px;
  margin-left: 30px;
}