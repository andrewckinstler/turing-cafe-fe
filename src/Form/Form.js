import React, { Component } from 'react';
import './Form.css'

export class Form extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      date: '',
      time: '',
      number: ''
    }
  }

  handleChange = e => {
    this.setState({ [e.target.name]: e.target.value})
  }

  resetInputs = () => {
    this.setState({ name: '', date: '', time: '', number: ''})
  }

  createReservation = () => {
    const { addReservation } = this.props
    const newReservation = { ...this.state, id: Date.now() }
    addReservation(newReservation)
    this.resetInputs() 
  }

  render() {
    const { name, date, time, number } = this.state;
    return (
      <div className='form-box'>
        <input
          type='text'
          placeholder='Name'
          value={name}
          name='name'
          onChange={this.handleChange}
        />
        <input
          type='text'
          placeholder='Date (mm/dd)'
          value={date}
          name='date'
          onChange={this.handleChange}
        />
        <input
          type='text'
          placeholder='Time'
          value={time}
          name='time'
          onChange={this.handleChange}
        />
        <input
          type='number'
          placeholder='Number of guests'
          value={number}
          name='number'
          onChange={this.handleChange}
        />
        <button 
          className='create'
          onClick={this.createReservation}>
            Create reservation
        </button>
      </div>
    )
  }
}