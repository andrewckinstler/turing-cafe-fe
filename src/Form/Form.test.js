import React from 'react';
import { shallow } from 'enzyme'
import { Form } from './Form';

it('should match a snapshot', () => {
  const wrapper = shallow(<Form />)
  expect(wrapper).toMatchSnapshot()
});

it('should change state when handleChange is called', () => {
  const wrapper = shallow(<Form />)
  const mockEvent = { target: { name: 'name', value: 'Sharon'} };
  const expected = 'Sharon';

  wrapper.instance().handleChange(mockEvent);

  expect(wrapper.state('name')).toEqual(expected);
});