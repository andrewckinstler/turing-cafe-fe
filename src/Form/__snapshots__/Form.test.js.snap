// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match a snapshot 1`] = `
<div>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    name="date"
    onChange={[Function]}
    placeholder="Date"
    type="date"
    value=""
  />
  <input
    name="time"
    onChange={[Function]}
    placeholder="Time"
    type="time"
    value=""
  />
  <input
    name="number"
    onChange={[Function]}
    placeholder="Number of guests"
    type="number"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Create reservation
  </button>
</div>
`;
